File: 9/perl/sample.pl
sub end {
  if ($_[0]>0) {
    print("".$_[0]."\n");
    end ($_[0]-1);
    return($_[0]);
  }
  print("end\n");
  return($_[0]);
}

sub start {
  print("start\n");
  while(1) {
    end($_[0]);
    sleep(1);
  }
}

start(3);

File: 9/java/Sample.java
public class Sample {
	public static int end(int value) {
		if (value > 0) {
			System.out.println(value);
			return end(value-1);
		}
		System.out.println("End");
		return value;
	}

	public static int start(int value) {
		while(true) {
			System.out.println("Start");
			end(value);
			try {
				Thread.sleep(1000);
			} catch (InterruptedException e) {
				System.err.format("%s\n", e);
			}
		}
	}

	public static void main(String []args) {
		System.out.println("Hello, world\n");
		start(3);
	}
}
